core/templates/pages/contributor-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -128,10 +132,15 @@ export class ContributionOpportunitiesBackendApiService {
   }
 
   async fetchTranslationOpportunitiesAsync(
-      languageCode: string, cursor: string):
+      languageCode: string, topicName: string, cursor: string):
     Promise<TranslationContributionOpportunities> {
+    if (topicName === 'All') {
+      topicName = '';
+    }
```

Comment:
Can we use a ternary instead in the params declaration below? Also can we pass null, as it seems more descriptive than an empty string.

Reply:
Could you explain what a ternary is?

I was using null initially, but the query it sent will be something like `/opportunitiessummaryhandler/translate/language_code=en&topic_name=null&cursor=`. Having `null` in the query is a little messy imo.

Reply:
@sophiewu6 does this help? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

I think either empty string or omitting the argument entirely is probably better since the arg needs to be of type string (if it exists). I lean a little bit towards the second, and treating it as an optional param in the backend. Would that work?

Reply:
Thanks for the suggestion! I personally think leaving an empty string would be slightly better because other params like `cursor` would have an empty string when we are not passing any cursor to the query.